mlir/test/Dialect/MIOpen/lowering_wrw_atomic_transform.mlir

```diff
@@ -7,8 +7,8 @@ module  {
-// CHECK: #transform_map{{[0-9]+}} = #miopen.transform_map<affine_map<(d0, d1, d2, d3, d4, d5) -> (d1 * 16 + d2, d0, d3, d4 - 2, d5 - 2)> by[#miopen.transform<PassThrough ["gi"] at [0] -> ["gi"] at [1]>, #miopen.transform<Unmerge{2, 16} ["n0", "n1"] at [1, 2] -> ["ni"] at [0]>, #miopen.transform<PassThrough ["ci"] at [3] -> ["ci"] at [2]>, #miopen.transform<Pad{2, 2, 2, 2} ["hipad", "wipad"] at [4, 5] -> ["hi", "wi"] at [3, 4]>] bounds = [1, 2, 16, 32, 11, 11] -> [32, 1, 32, 7, 7]>
-// CHECK: #transform_map{{[0-9]+}} = #miopen.transform_map<affine_map<(d0, d1, d2, d3, d4, d5, d6, d7) -> (d0, d1, d2, d3, d4 + d5, d6 + d7)> by[#miopen.transform<PassThrough ["gi"] at [0] -> ["gi"] at [0]>, #miopen.transform<PassThrough ["n0"] at [1] -> ["n0"] at [1]>, #miopen.transform<PassThrough ["n1"] at [2] -> ["n1"] at [2]>, #miopen.transform<PassThrough ["ci"] at [3] -> ["ci"] at [3]>, #miopen.transform<Embed{1, 1} ["y", "ho"] at [4, 5] -> ["hipad"] at [4]>, #miopen.transform<Embed{1, 1} ["x", "wo"] at [6, 7] -> ["wipad"] at [5]>] bounds = [1, 2, 16, 32, 3, 9, 3, 9] -> [1, 2, 16, 32, 11, 11]>
-// CHECK: #transform_map{{[0-9]+}} = #miopen.transform_map<affine_map<(d0, d1, d2) -> (d0 floordiv 2, d0 mod 2, d1 floordiv 81, d2 floordiv 9, (d2 mod 9) floordiv 3, (d1 mod 81) floordiv 9, d2 mod 3, d1 mod 9)> by[#miopen.transform<Merge{1, 2} ["gemmG"] at [0] -> ["gi", "n0"] at [0, 1]>, #miopen.transform<Merge{16, 9, 9} ["gemmK"] at [1] -> ["n1", "ho", "wo"] at [2, 5, 7]>, #miopen.transform<Merge{32, 3, 3} ["gemmN"] at [2] -> ["ci", "y", "x"] at [3, 4, 6]>] bounds = [2, 1296, 288] -> [1, 2, 16, 32, 3, 9, 3, 9]>
-// CHECK: #transform_map{{[0-9]+}} = #miopen.transform_map<affine_map<(d0, d1, d2, d3, d4, d5) -> (d1 * 16 + d2, d0, d3, d4, d5)> by[#miopen.transform<PassThrough ["go"] at [0] -> ["go"] at [1]>, #miopen.transform<Unmerge{2, 16} ["n0", "n1"] at [1, 2] -> ["no"] at [0]>, #miopen.transform<PassThrough ["ko", "ho", "wo"] at [3, 4, 5] -> ["ko", "ho", "wo"] at [2, 3, 4]>] bounds = [1, 2, 16, 32, 9, 9] -> [32, 1, 32, 9, 9]>
-// CHECK: #transform_map{{[0-9]+}} = #miopen.transform_map<affine_map<(d0, d1, d2) -> (d0 floordiv 2, d0 mod 2, d1 floordiv 81, d2, (d1 mod 81) floordiv 9, d1 mod 9)> by[#miopen.transform<Merge{1, 2} ["gemmG"] at [0] -> ["go", "n0"] at [0, 1]>, #miopen.transform<Merge{16, 9, 9} ["gemmK"] at [1] -> ["n1", "ho", "wo"] at [2, 4, 5]>, #miopen.transform<PassThrough ["gemmM"] at [2] -> ["ko"] at [3]>] bounds = [2, 1296, 32] -> [1, 2, 16, 32, 9, 9]>
+// CHECK: #transform_map{{[0-9]+}} = #miopen.transform_map<affine_map<(d0, d1, d2, d3, d4, d5) -> (d0 * 16 + d1, d2, d3, d4 - 2, d5 - 2)> by[#miopen.transform<PassThrough ["gi"] at [2] -> ["gi"] at [1]>, #miopen.transform<Unmerge{2, 16} ["n0", "n1"] at [0, 1] -> ["ni"] at [0]>, #miopen.transform<PassThrough ["ci"] at [3] -> ["ci"] at [2]>, #miopen.transform<Pad{2, 2, 2, 2} ["hipad", "wipad"] at [4, 5] -> ["hi", "wi"] at [3, 4]>] bounds = [2, 16, 1, 32, 11, 11] -> [32, 1, 32, 7, 7]>
```

Comment:
the unit test here only ues NCHW, can we add another NHWC case, so perhaps it can be observed from the `#transform_map` attributes that a simpler transformation is derived?

Reply:
Done